### PR TITLE
feat: drop support for warehouse time intervals

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #3184 

### Description:

Changes:
- drop support for custom strings in the time_intervals
- update docs with supported options -> https://deploy-preview-3306--peaceful-bassi-cbf284.netlify.app/references/dimensions#options
- initial setup of `timeFrameConfigs` where we will centralise all the logic and configuration around time frames. Inspired by https://medium.com/thousandeyes-engineering/the-distributed-enum-anti-pattern-3ebb23cbc5d8 ( following PR's will take advantage of this )

### CODE REVIEW

**Review by rephus — APPROVED** [review]
Looks good, this should not break the current support for `HOUR` like the one I have on my schema.yml, right ? 

image

**Comment by ZeRego** [comment]
> Looks good, this should not break the current support for `HOUR` like the one I have on my schema.yml, right ?
> 
> image

You'r good 👍 I kept support to all the values we had in our code which included HOUR.

All the supported options -> https://deploy-preview-3306--peaceful-bassi-cbf284.netlify.app/references/dimensions#options

File: docs/docs/references/dimensions.mdx [hunk 1]
```diff
@@ -120,29 +120,6 @@ Lightdash breaks this out into the default intervals automatically. So, this is
 **Date**: ['DAY', 'WEEK', 'MONTH', 'YEAR']
```
Comment (TuringLovesDeathMetal) [inline docs/docs/references/dimensions.mdx:120]:
```suggestion
**Date**: ['DAY', 'WEEK', 'MONTH', 'YEAR']

```

Comment (TuringLovesDeathMetal) [reply docs/docs/references/dimensions.mdx:120]: For some reason, these appear on the same line in the docs

**Review by TuringLovesDeathMetal — COMMENTED** [review]
One tiny change to the docs (that wasn't actually your doing 😅 )

Also - I'm sure you've thought of this already - but is this backwards-compatible with the current timeframes that we support?

**Pull request merged**